src/Tutorials/Unit-Testing-Grains.md

```diff
@@ -5,81 +5,159 @@ title: Unit Testing Grains
 
 # Unit Testing Grains
 
-This tutorial shows you how to write unit tests for your grains to make sure they are behaving correctly.
-For a distributed application written in Orleans, you'll need load tests and integration tests as well but here we'll only focus on unit tests.
+This tutorial shows how to unit test your grains to make sure they behave correctly.
+There are two main ways to unit test your grains, and the method you choose will depend on the type of functionality you are testing.
+You can use a mocking framework like [Moq](https://github.com/moq/moq) to mock parts of the Orleans runtime that your grain interacts with, or you can use the  `Microsoft.Orleans.TestingHost` NuGet package to create test silos for your grains.
 
-Orleans makes it possible to mock many of its parts ([example](https://github.com/dotnet/orleans/tree/master/Samples/UnitTesting.Minimal)) but here we only focus on simply running grains in test silos.
+## Using Mocks
 
-The steps are
+Orleans makes it possible to mock many parts of system, and for a lot of scenarios this is the easiest way to unit test grains. 
+This approach does have limitations, and may require that grains include code used only by your unit tests.
```

Comment:
This is not the only limitation. It has several, especially with regards to scheduling, reentrancy, serialization, which will behave very differently in the silo than directly on the framework. I would highlight this.